.github/workflows/cpp-packaging.yml

```diff
@@ -35,9 +39,26 @@ env:
   xcodeVersion: "12"
 
 jobs:
+  prepare_matrix:
```

Comment:
So, I believe we should actually *use* the prepared matrix on the "tests" job near the bottom of this file. I'd rather not leave this for a subsequent PR.